Core/clim-basic/extended-streams/text-formatting.lisp

```diff
@@ -44,7 +44,9 @@
                   (setf (getf text-margins ,edge) ,default))))
     (thunk :left   '(:absolute 0))
     (thunk :top    '(:absolute 0))
-    (thunk :right  (or text-margin '(:relative 0)))
+    (thunk :right  (if text-margin
+                       `(:absolute ,text-margin)
+                       `(:relative 0)))
     (thunk :bottom '(:relative 0))
```

Comment:
I suggest extracting this logic into a function `(normalize-text-margins text-margins text-margin)` or similar and adding a unit test.

Reply:
this logic with default value is used only in this particular method (there is something similar but not quite the same in setf stream-text-margins) -- is there much of a value in extracting it?